snapcraft/internal/lxd/_cleanbuilder.py

```diff
@@ -35,8 +36,11 @@ def __init__(self, *, output=None, source, project_options,
                          container_name=container_name, remote=remote)
 
     def _ensure_container(self):
-        subprocess.check_call([
-            'lxc', 'launch', '-e', self._image, self._container_name])
+        try:
+            subprocess.check_call([
+                'lxc', 'launch', '-e', self._image, self._container_name])
+        except subprocess.CalledProcessError as e:
+            raise ContainerError(e)
```

Comment:
you want this to be `ContainerSnapcraftCmdError` which is more correct and silences the exception printing of `cmd`

Reply:
That can't be the right exception, it's message is "Snapcraft command failed in the container:". Do you mean ContainerRunError? (I find over-elaborate exception hierarchies a bit of an anti-pattern myself...)

Reply:
In my similar PR #1761 I used `ContainerConnectionError('Failed to setup container')` because it will refer the user to online docs for LXD, which I think is helpful when you get something like "No storage pool found. Please create a new storage pool." and don't know what that means. So personally I think we want that message here.

What @sergiusens is suggesting is to use `ContainerSnapcraftCmdError` because it won't add anything on top of what the `lxc` command may have printed. I also think that's the wrong error because it's specifically referring to running Snapcraft inside the container - that's not the case here.
If we wanted to go that route I think it's time we moved `should_print_error` into the `SnapcraftError` class to be used with a new exception here.

Reply:
I don't want to be a jerk but at this stage given how simple the change is, maybe it would be easier for one of you to JFDI this?

Reply:
Done